plugins/lighthouse/src/components/AuditView/index.test.tsx

```diff
@@ -0,0 +1,226 @@
+      });
+    });
+
+    it('sets the current audit as active', async () => {
+      const rendered = render(
+        wrapInThemedTestApp(
+          <ApiProvider apis={apis}>
+            <AuditView />
+          </ApiProvider>,
+        ),
+      );
+
+      await rendered.findByTestId('audit-sidebar');
+
+      const audit = website.audits.find(a => a.id === id) as Audit;
+      const auditElement = rendered.getByText(formatTime(audit.timeCreated));
+      expect(auditElement.parentElement?.parentElement?.className).toContain(
+        'selected',
+      );
+
+      const notSelectedAudit = website.audits.find(a => a.id !== id) as Audit;
+      const notSelectedAuditElement = rendered.getByText(
+        formatTime(notSelectedAudit.timeCreated),
+      );
+      expect(
+        notSelectedAuditElement.parentElement?.parentElement?.className,
+      ).not.toContain('selected');
+    });
+
+    it('navigates to the next report when an audit is clicked', async () => {
+      const rendered = render(
+        wrapInThemedTestApp(
+          <ApiProvider apis={apis}>
+            <AuditView />
+          </ApiProvider>,
+        ),
+      );
+
+      await rendered.findByTestId('audit-sidebar');
+
+      website.audits.forEach(a => {
+        expect(
+          rendered.getByText(formatTime(a.timeCreated)).parentElement
+            ?.parentElement,
+        ).toHaveAttribute('href', `/lighthouse/audit/${a.id}`);
+      });
+    });
+  });
+
+  describe('when the request for the website by id is pending', () => {
+    it('it shows the loading', async () => {
+      mockFetch.mockImplementationOnce(() => new Promise(() => {}));
+      const rendered = render(
+        wrapInThemedTestApp(
+          <ApiProvider apis={apis}>
+            <AuditView />
+          </ApiProvider>,
+        ),
+      );
+      expect(await rendered.findByTestId('progress')).toBeInTheDocument();
+    });
+  });
+
+  describe('when the request for the website by id fails', () => {
+    it('it shows an error', async () => {
+      mockFetch.mockRejectOnce(new Error('failed to fetch'));
+      const rendered = render(
+        wrapInThemedTestApp(
+          <ApiProvider apis={apis}>
+            <AuditView />
+          </ApiProvider>,
+        ),
+      );
+      expect(await rendered.findByText('failed to fetch')).toBeInTheDocument();
+    });
+  });
+
+  describe.skip('when a loading audit is accessed', () => {
+    it('shows a loading view', async () => {
+      id = website.audits.find(a => a.status === 'RUNNING')?.id as string;
+      useParams.mockReturnValueOnce({ id });
+
+      const rendered = render(
+        wrapInThemedTestApp(
+          <ApiProvider apis={apis}>
+            <AuditView />
+          </ApiProvider>,
+        ),
+      );
+
+      await rendered.findByTestId('audit-sidebar');
+
+      expect(rendered.queryByTestId('progress')).toBeInTheDocument();
+    });
+  });
+
+  describe.skip('when a failed audit is accessed', () => {
```

Comment:
full disclosure: not really sure how to test this, so it is skipped. maybe I fish around and use expect(HTMLElement).toHaveAttributes(), but it'd be better as a viz regression test with storybook or something.